tests/integration/test_self_hosted_runner.py

```diff
@@ -92,3 +95,87 @@ async def test_dispatch_workflow_with_dockerhub_mirror(
                 "A private docker registry is setup as a dockerhub mirror for this self-hosted"
                 " runner."
             ) in logs
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_flush_busy_runner(
+    model: Model,
+    app_runner: Application,
+    forked_github_repository: Repository,
+    runner_manager_github_client: GithubClient,
+) -> None:
+    """
+    arrange: A working application with one runner.
+    act:
+        1. Dispatch a workflow that waits for 30 mins.
+        2. Run flush-runners action.
+    assert:
+        1. The runner is in busy status.
+        2.  a. The flush-runners action should take less than the timeout.
+            b. The runner should be flushed.
+    """
+    unit = app_runner.units[0]
+
+    config = await app_runner.get_config()
+
+    await app_runner.set_config(
+        {"path": forked_github_repository.full_name, "virtual-machines": "1"}
+    )
+    await reconcile(app=app_runner, model=model)
+    await wait_till_num_of_runners(unit, 1)
+
+    names = await get_runner_names(unit)
+    assert len(names) == 1
+
+    runner_to_be_used = names[0]
+
+    # 1.
+    main_branch = forked_github_repository.get_branch(forked_github_repository.default_branch)
+    workflow = forked_github_repository.get_workflow(
+        id_or_file_name=DISPATCH_WAIT_TEST_WORKFLOW_FILENAME
+    )
+
+    assert workflow.create_dispatch(main_branch, {"runner": app_runner.name, "minutes": "30"})
+
+    # Wait until runner online and then busy.
+    for _ in range(30):
+        all_runners = runner_manager_github_client.get_runner_github_info(
+            GithubRepo(
+                owner=forked_github_repository.owner.login, repo=forked_github_repository.name
+            )
+        )
+        runners = [runner for runner in all_runners if runner.name == runner_to_be_used]
+
+        if not runners:
+            # if runner is not online yet.
+            sleep(30)
+            continue
```

Comment:
I'm wondering if this could benefit in readability by using `helpers.wait_for` method.

Reply:
The same with timeout errors below.

Reply:
For this one, I feel it might be clearer like this.

The alternative is to have line 143 - 159 as a function the `wait_for` .
This would involve using `functools.partial` or defining a inner function in the scope to load the `runner_to_be_used` to that function.

I feel that might be over complexing things.